Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -486,6 +496,86 @@ final class PurchasesOrchestrator {
         }
     }
 
+    // Win-back offers weren't introduced until iOS 18 and Xcode 16, which shipped with version 6.0 of the Swift
+    // compiler. The win-back symbols won't be found if compiled on Xcode < 16.0, so we need to ensure that
+    // the Swift compiler 6.0 or higher is available.
+    #if compiler(>=6.0)
+    @available(iOS 18.0, macOS 15.0, tvOS 18.0, watchOS 11.0, visionOS 2.0, *)
```

Comment:
This method feels like it has a lot of duplication with the other purchase method that includes an optional `promotionalOffer` parameter. What do you think about adding `winBackOffer: Product.SubscriptionOffer` as an optional parameter there, and we can use the compiler flag check for just the usage of `.winBackOffer(winBackOffer)` inside? I believe that is the only symbol that requires the compiler check since `Product.SubscriptionOffer` is available since iOS 15.

```
#if compiler(>=6.0)
options.insert(.winBackOffer(winBackOffer))
#endif
```

Reply:
I think this makes a lot of sense. I've made those changes in https://github.com/RevenueCat/purchases-ios/pull/4370/commits/bbeb427c7690e6c2f95f73f2b3cbcf9d3b9e59ad. I think we'll just need to be careful not to pass in both a promotional offer and a win-back offer at the same time, but it's difficult to imagine a scenario where that'd happen, especially since this is an internal SDK function